metabase/metabase
# Add engine search to the database form

Epic https://github.com/metabase/metabase/issues/26178

Changes:
- Adds a way to set initial values for the form
- Adds a concept of `advanced` mode, where in non-advanced mode some fields are hidden (porting existing behavior)
- Adds a widget to search engines when in the non-advanced mode (porting existing behavior). I had to **copy-paste** code for it because it relied on the old form framework, but I'll remove this duplication in ongoing PRs.
- Please note that logos won't work in storybook

How to test:
- Copy-paste `engines` setting value to `DatabaseForm.stories.tsx`
- Make sure the form works in the non-advanced mode (default one)

<img width="700" alt="Screenshot 2022-11-29 at 19 48 51" src="https://user-images.githubusercontent.com/8542534/204605165-da6b2fc3-560f-4205-9a88-a5611667c2e8.png">
<img width="705" alt="Screenshot 2022-11-29 at 19 49 03" src="https://user-images.githubusercontent.com/8542534/204605175-924f7879-db8e-4382-a803-cfd1a8ff3abd.png">
